// @aws-sdk/client-textract/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, AwsSdkSigV4Signer, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsJson1_1Protocol } = require("@aws-sdk/core/protocols");
const { Sha256 } = require("@smithy/core/checksum");

const defaultTextractHttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "textract",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const defaultTextractHttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
        }
    }
    return options;
};
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    return Object.assign(config_0, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "textract",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1078.0";
var packageInfo = {
	version: version};

const k = "ref";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "getAttr", g = { [k]: "Endpoint" }, h = { [k]: d }, i = {}, j = [{ [k]: "Region" }];
const _data = {
    conditions: [
        [c, [g]],
        [c, j],
        ["aws.partition", j, d],
        [e, [{ [k]: "UseFIPS" }, b]],
        [e, [{ [k]: "UseDualStack" }, b]],
        [e, [{ fn: f, argv: [h, "supportsDualStack"] }, b]],
        [e, [{ fn: f, argv: [h, "supportsFIPS"] }, b]]
    ],
    results: [
        [a],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [g, i],
        ["https://textract-fips.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://textract-fips.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://textract.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://textract.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 12, 3,
    1, 4, r + 11,
    2, 5, r + 11,
    3, 8, 6,
    4, 7, r + 10,
    5, r + 8, r + 9,
    4, 10, 9,
    6, r + 6, r + 7,
    5, 11, r + 5,
    6, r + 4, r + 5,
    3, r + 1, 13,
    4, r + 2, r + 3,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

class TextractServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, TextractServiceException.prototype);
    }
}

class AccessDeniedException extends TextractServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class BadDocumentException extends TextractServiceException {
    name = "BadDocumentException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "BadDocumentException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadDocumentException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class DocumentTooLargeException extends TextractServiceException {
    name = "DocumentTooLargeException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "DocumentTooLargeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DocumentTooLargeException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class HumanLoopQuotaExceededException extends TextractServiceException {
    name = "HumanLoopQuotaExceededException";
    $fault = "client";
    ResourceType;
    QuotaCode;
    ServiceCode;
    Message;
    Code;
    constructor(opts) {
        super({
            name: "HumanLoopQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HumanLoopQuotaExceededException.prototype);
        this.ResourceType = opts.ResourceType;
        this.QuotaCode = opts.QuotaCode;
        this.ServiceCode = opts.ServiceCode;
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class InternalServerError extends TextractServiceException {
    name = "InternalServerError";
    $fault = "server";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "InternalServerError",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerError.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class InvalidParameterException extends TextractServiceException {
    name = "InvalidParameterException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "InvalidParameterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidParameterException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class InvalidS3ObjectException extends TextractServiceException {
    name = "InvalidS3ObjectException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "InvalidS3ObjectException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidS3ObjectException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ProvisionedThroughputExceededException extends TextractServiceException {
    name = "ProvisionedThroughputExceededException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ProvisionedThroughputExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ProvisionedThroughputExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ThrottlingException extends TextractServiceException {
    name = "ThrottlingException";
    $fault = "server";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class UnsupportedDocumentException extends TextractServiceException {
    name = "UnsupportedDocumentException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "UnsupportedDocumentException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedDocumentException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ConflictException extends TextractServiceException {
    name = "ConflictException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class IdempotentParameterMismatchException extends TextractServiceException {
    name = "IdempotentParameterMismatchException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "IdempotentParameterMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IdempotentParameterMismatchException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class LimitExceededException extends TextractServiceException {
    name = "LimitExceededException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ServiceQuotaExceededException extends TextractServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ValidationException extends TextractServiceException {
    name = "ValidationException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class InvalidKMSKeyException extends TextractServiceException {
    name = "InvalidKMSKeyException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "InvalidKMSKeyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKMSKeyException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class ResourceNotFoundException extends TextractServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
class InvalidJobIdException extends TextractServiceException {
    name = "InvalidJobIdException";
    $fault = "client";
    Message;
    Code;
    constructor(opts) {
        super({
            name: "InvalidJobIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidJobIdException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}

const _A = "Adapter";
const _AC = "AdaptersConfig";
const _ACT = "AfterCreationTime";
const _AD = "AnalyzeDocument";
const _ADE = "AccessDeniedException";
const _ADMV = "AnalyzeDocumentModelVersion";
const _ADR = "AnalyzeDocumentRequest";
const _ADRn = "AnalyzeDocumentResponse";
const _AE = "AnalyzeExpense";
const _AEMV = "AnalyzeExpenseModelVersion";
const _AER = "AnalyzeExpenseRequest";
const _AERn = "AnalyzeExpenseResponse";
const _AI = "AdapterId";
const _AID = "AnalyzeID";
const _AIDD = "AnalyzeIDDetections";
const _AIDMV = "AnalyzeIDModelVersion";
const _AIDR = "AnalyzeIDRequest";
const _AIDRn = "AnalyzeIDResponse";
const _AL = "AdapterList";
const _ALMV = "AnalyzeLendingModelVersion";
const _AN = "AdapterName";
const _AO = "AdapterOverview";
const _AU = "AutoUpdate";
const _AV = "AdapterVersion";
const _AVDC = "AdapterVersionDatasetConfig";
const _AVEM = "AdapterVersionEvaluationMetric";
const _AVEMd = "AdapterVersionEvaluationMetrics";
const _AVL = "AdapterVersionList";
const _AVO = "AdapterVersionOverview";
const _AVd = "AdapterVersions";
const _Ad = "Adapters";
const _Al = "Alias";
const _B = "Baseline";
const _BB = "BoundingBox";
const _BCT = "BeforeCreationTime";
const _BDE = "BadDocumentException";
const _BL = "BlockList";
const _BT = "BlockType";
const _Bl = "Blocks";
const _Blo = "Block";
const _Bu = "Bucket";
const _By = "Bytes";
const _C = "Code";
const _CA = "CreateAdapter";
const _CAR = "CreateAdapterRequest";
const _CARr = "CreateAdapterResponse";
const _CAV = "CreateAdapterVersion";
const _CAVR = "CreateAdapterVersionRequest";
const _CAVRr = "CreateAdapterVersionResponse";
const _CC = "ContentClassifiers";
const _CE = "ConflictException";
const _CI = "ColumnIndex";
const _CRT = "ClientRequestToken";
const _CS = "ColumnSpan";
const _CT = "CreationTime";
const _Co = "Confidence";
const _Cu = "Currency";
const _D = "Document";
const _DA = "DataAttributes";
const _DAR = "DeleteAdapterRequest";
const _DARe = "DeleteAdapterResponse";
const _DAV = "DeleteAdapterVersion";
const _DAVR = "DeleteAdapterVersionRequest";
const _DAVRe = "DeleteAdapterVersionResponse";
const _DAe = "DeleteAdapter";
const _DC = "DatasetConfig";
const _DDT = "DetectDocumentText";
const _DDTMV = "DetectDocumentTextModelVersion";
const _DDTR = "DetectDocumentTextRequest";
const _DDTRe = "DetectDocumentTextResponse";
const _DG = "DocumentGroup";
const _DGL = "DocumentGroupList";
const _DGo = "DocumentGroups";
const _DI = "DocumentIndex";
const _DL = "DocumentLocation";
const _DM = "DocumentMetadata";
const _DP = "DocumentPages";
const _DS = "DetectedSignature";
const _DSL = "DetectedSignatureList";
const _DSe = "DetectedSignatures";
const _DTLE = "DocumentTooLargeException";
const _De = "Description";
const _E = "Extraction";
const _EC = "ExpenseCurrency";
const _ECr = "ErrorCode";
const _ED = "ExpenseDocuments";
const _EDL = "ExpenseDocumentList";
const _EDx = "ExpenseDetection";
const _EDxp = "ExpenseDocument";
const _EF = "ExpenseField";
const _EFL = "ExpenseFieldList";
const _EGP = "ExpenseGroupProperty";
const _EGPL = "ExpenseGroupPropertyList";
const _EI = "ExpenseIndex";
const _EL = "ExtractionList";
const _EM = "EvaluationMetric";
const _EMv = "EvaluationMetrics";
const _ET = "EntityTypes";
const _ETx = "ExpenseType";
const _Ex = "Extractions";
const _FDA = "FlowDefinitionArn";
const _FS = "F1Score";
const _FT = "FeatureTypes";
const _FTe = "FeatureType";
const _G = "Geometry";
const _GA = "GetAdapter";
const _GAR = "GetAdapterRequest";
const _GARe = "GetAdapterResponse";
const _GAV = "GetAdapterVersion";
const _GAVR = "GetAdapterVersionRequest";
const _GAVRe = "GetAdapterVersionResponse";
const _GDA = "GetDocumentAnalysis";
const _GDAR = "GetDocumentAnalysisRequest";
const _GDARe = "GetDocumentAnalysisResponse";
const _GDTD = "GetDocumentTextDetection";
const _GDTDR = "GetDocumentTextDetectionRequest";
const _GDTDRe = "GetDocumentTextDetectionResponse";
const _GEA = "GetExpenseAnalysis";
const _GEAR = "GetExpenseAnalysisRequest";
const _GEARe = "GetExpenseAnalysisResponse";
const _GLA = "GetLendingAnalysis";
const _GLAR = "GetLendingAnalysisRequest";
const _GLARe = "GetLendingAnalysisResponse";
const _GLAS = "GetLendingAnalysisSummary";
const _GLASR = "GetLendingAnalysisSummaryRequest";
const _GLASRe = "GetLendingAnalysisSummaryResponse";
const _GP = "GroupProperties";
const _H = "Height";
const _HLA = "HumanLoopArn";
const _HLACER = "HumanLoopActivationConditionsEvaluationResults";
const _HLAO = "HumanLoopActivationOutput";
const _HLAR = "HumanLoopActivationReasons";
const _HLC = "HumanLoopConfig";
const _HLDA = "HumanLoopDataAttributes";
const _HLN = "HumanLoopName";
const _HLQEE = "HumanLoopQuotaExceededException";
const _I = "Id";
const _ID = "IdentityDocuments";
const _IDF = "IdentityDocumentFields";
const _IDFL = "IdentityDocumentFieldList";
const _IDFd = "IdentityDocumentField";
const _IDL = "IdentityDocumentList";
const _IDd = "IdentityDocument";
const _IJIE = "InvalidJobIdException";
const _IKMSKE = "InvalidKMSKeyException";
const _IPE = "InvalidParameterException";
const _IPME = "IdempotentParameterMismatchException";
const _ISE = "InternalServerError";
const _ISOE = "InvalidS3ObjectException";
const _Id = "Ids";
const _In = "Index";
const _JI = "JobId";
const _JS = "JobStatus";
const _JT = "JobTag";
const _KD = "KeyDetection";
const _KMSKI = "KMSKeyId";
const _L = "Left";
const _LA = "ListAdapters";
const _LAR = "ListAdaptersRequest";
const _LARi = "ListAdaptersResponse";
const _LAV = "ListAdapterVersions";
const _LAVR = "ListAdapterVersionsRequest";
const _LAVRi = "ListAdapterVersionsResponse";
const _LD = "LabelDetection";
const _LDL = "LendingDetectionList";
const _LDe = "LendingDocument";
const _LDen = "LendingDetection";
const _LEE = "LimitExceededException";
const _LF = "LendingFields";
const _LFL = "LendingFieldList";
const _LFe = "LendingField";
const _LI = "LineItems";
const _LIEF = "LineItemExpenseFields";
const _LIF = "LineItemFields";
const _LIG = "LineItemGroups";
const _LIGI = "LineItemGroupIndex";
const _LIGL = "LineItemGroupList";
const _LIGi = "LineItemGroup";
const _LIL = "LineItemList";
const _LR = "LendingResult";
const _LRL = "LendingResultList";
const _LS = "LendingSummary";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _M = "Message";
const _MR = "MaxResults";
const _MSO = "ManifestS3Object";
const _N = "Name";
const _NC = "NotificationChannel";
const _NT = "NextToken";
const _NV = "NormalizedValue";
const _OC = "OutputConfig";
const _P = "Pages";
const _PC = "PageClassification";
const _PL = "PredictionList";
const _PN = "PageNumber";
const _PT = "PageType";
const _PTEE = "ProvisionedThroughputExceededException";
const _Pa = "Page";
const _Po = "Polygon";
const _Poi = "Point";
const _Pr = "Precision";
const _Pre = "Prediction";
const _Q = "Query";
const _QC = "QuotaCode";
const _QCu = "QueriesConfig";
const _Qu = "Queries";
const _R = "Relationships";
const _RA = "RotationAngle";
const _RARN = "ResourceARN";
const _RAo = "RoleArn";
const _RI = "RowIndex";
const _RL = "RelationshipList";
const _RNFE = "ResourceNotFoundException";
const _RS = "RowSpan";
const _RT = "ResourceType";
const _Re = "Recall";
const _Rel = "Relationship";
const _Res = "Results";
const _S = "Status";
const _SB = "S3Bucket";
const _SC = "ServiceCode";
const _SD = "SplitDocuments";
const _SDA = "StartDocumentAnalysis";
const _SDAR = "StartDocumentAnalysisRequest";
const _SDARt = "StartDocumentAnalysisResponse";
const _SDL = "SignatureDetectionList";
const _SDLp = "SplitDocumentList";
const _SDTD = "StartDocumentTextDetection";
const _SDTDR = "StartDocumentTextDetectionRequest";
const _SDTDRt = "StartDocumentTextDetectionResponse";
const _SDi = "SignatureDetections";
const _SDig = "SignatureDetection";
const _SDp = "SplitDocument";
const _SEA = "StartExpenseAnalysis";
const _SEAR = "StartExpenseAnalysisRequest";
const _SEARt = "StartExpenseAnalysisResponse";
const _SF = "SummaryFields";
const _SJHLACER = "SynthesizedJsonHumanLoopActivationConditionsEvaluationResults";
const _SLA = "StartLendingAnalysis";
const _SLAR = "StartLendingAnalysisRequest";
const _SLARt = "StartLendingAnalysisResponse";
const _SM = "StatusMessage";
const _SNSTA = "SNSTopicArn";
const _SO = "S3Object";
const _SP = "S3Prefix";
const _SQEE = "ServiceQuotaExceededException";
const _SS = "SelectionStatus";
const _Su = "Summary";
const _T = "Text";
const _TE = "ThrottlingException";
const _TK = "TagKeys";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TT = "TextType";
const _Ta = "Tags";
const _To = "Top";
const _Ty = "Type";
const _Typ = "Types";
const _UA = "UpdateAdapter";
const _UAR = "UpdateAdapterRequest";
const _UARp = "UpdateAdapterResponse";
const _UDE = "UnsupportedDocumentException";
const _UDT = "UndetectedDocumentTypes";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _US = "UndetectedSignatures";
const _USL = "UndetectedSignatureList";
const _USn = "UndetectedSignature";
const _V = "Version";
const _VD = "ValueDetection";
const _VDa = "ValueDetections";
const _VE = "ValidationException";
const _VT = "ValueType";
const _Va = "Value";
const _W = "Width";
const _Wa = "Warnings";
const _War = "Warning";
const _X = "X";
const _Y = "Y";
const _a = "application/json";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _mT = "mediaType";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.textract";
const _se = "server";
const n0 = "com.amazonaws.textract";
const _s_registry = TypeRegistry.for(_s);
var TextractServiceException$ = [-3, _s, "TextractServiceException", 0, [], []];
_s_registry.registerError(TextractServiceException$, TextractServiceException);
const n0_registry = TypeRegistry.for(n0);
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
var BadDocumentException$ = [-3, n0, _BDE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(BadDocumentException$, BadDocumentException);
var ConflictException$ = [-3, n0, _CE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ConflictException$, ConflictException);
var DocumentTooLargeException$ = [-3, n0, _DTLE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(DocumentTooLargeException$, DocumentTooLargeException);
var HumanLoopQuotaExceededException$ = [-3, n0, _HLQEE,
    { [_e]: _c, [_hE]: 402 },
    [_RT, _QC, _SC, _M, _C],
    [0, 0, 0, 0, 0]
];
n0_registry.registerError(HumanLoopQuotaExceededException$, HumanLoopQuotaExceededException);
var IdempotentParameterMismatchException$ = [-3, n0, _IPME,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(IdempotentParameterMismatchException$, IdempotentParameterMismatchException);
var InternalServerError$ = [-3, n0, _ISE,
    { [_e]: _se },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(InternalServerError$, InternalServerError);
var InvalidJobIdException$ = [-3, n0, _IJIE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(InvalidJobIdException$, InvalidJobIdException);
var InvalidKMSKeyException$ = [-3, n0, _IKMSKE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(InvalidKMSKeyException$, InvalidKMSKeyException);
var InvalidParameterException$ = [-3, n0, _IPE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(InvalidParameterException$, InvalidParameterException);
var InvalidS3ObjectException$ = [-3, n0, _ISOE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(InvalidS3ObjectException$, InvalidS3ObjectException);
var LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(LimitExceededException$, LimitExceededException);
var ProvisionedThroughputExceededException$ = [-3, n0, _PTEE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ProvisionedThroughputExceededException$, ProvisionedThroughputExceededException);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
var ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ServiceQuotaExceededException$, ServiceQuotaExceededException);
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _se },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ThrottlingException$, ThrottlingException);
var UnsupportedDocumentException$ = [-3, n0, _UDE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(UnsupportedDocumentException$, UnsupportedDocumentException);
var ValidationException$ = [-3, n0, _VE,
    { [_e]: _c },
    [_M, _C],
    [0, 0]
];
n0_registry.registerError(ValidationException$, ValidationException);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var SynthesizedJsonHumanLoopActivationConditionsEvaluationResults = [0, n0, _SJHLACER, { [_mT]: _a }, 0];
var Adapter$ = [3, n0, _A,
    0,
    [_AI, _V, _P],
    [0, 0, 64 | 0], 2
];
var AdapterOverview$ = [3, n0, _AO,
    0,
    [_AI, _AN, _CT, _FT],
    [0, 0, 4, 64 | 0]
];
var AdaptersConfig$ = [3, n0, _AC,
    0,
    [_Ad],
    [() => Adapters], 1
];
var AdapterVersionDatasetConfig$ = [3, n0, _AVDC,
    0,
    [_MSO],
    [() => S3Object$]
];
var AdapterVersionEvaluationMetric$ = [3, n0, _AVEM,
    0,
    [_B, _AV, _FTe],
    [() => EvaluationMetric$, () => EvaluationMetric$, 0]
];
var AdapterVersionOverview$ = [3, n0, _AVO,
    0,
    [_AI, _AV, _CT, _FT, _S, _SM],
    [0, 0, 4, 64 | 0, 0, 0]
];
var AnalyzeDocumentRequest$ = [3, n0, _ADR,
    0,
    [_D, _FT, _HLC, _QCu, _AC],
    [() => Document$, 64 | 0, () => HumanLoopConfig$, () => QueriesConfig$, () => AdaptersConfig$], 2
];
var AnalyzeDocumentResponse$ = [3, n0, _ADRn,
    0,
    [_DM, _Bl, _HLAO, _ADMV],
    [() => DocumentMetadata$, () => BlockList, [() => HumanLoopActivationOutput$, 0], 0]
];
var AnalyzeExpenseRequest$ = [3, n0, _AER,
    0,
    [_D],
    [() => Document$], 1
];
var AnalyzeExpenseResponse$ = [3, n0, _AERn,
    0,
    [_DM, _ED],
    [() => DocumentMetadata$, () => ExpenseDocumentList]
];
var AnalyzeIDDetections$ = [3, n0, _AIDD,
    0,
    [_T, _NV, _Co],
    [0, () => NormalizedValue$, 1], 1
];
var AnalyzeIDRequest$ = [3, n0, _AIDR,
    0,
    [_DP],
    [() => DocumentPages], 1
];
var AnalyzeIDResponse$ = [3, n0, _AIDRn,
    0,
    [_ID, _DM, _AIDMV],
    [() => IdentityDocumentList, () => DocumentMetadata$, 0]
];
var Block$ = [3, n0, _Blo,
    0,
    [_BT, _Co, _T, _TT, _RI, _CI, _RS, _CS, _G, _I, _R, _ET, _SS, _Pa, _Q],
    [0, 1, 0, 0, 1, 1, 1, 1, () => Geometry$, 0, () => RelationshipList, 64 | 0, 0, 1, () => Query$]
];
var BoundingBox$ = [3, n0, _BB,
    0,
    [_W, _H, _L, _To],
    [1, 1, 1, 1]
];
var CreateAdapterRequest$ = [3, n0, _CAR,
    0,
    [_AN, _FT, _CRT, _De, _AU, _Ta],
    [0, 64 | 0, [0, 4], 0, 0, 128 | 0], 2
];
var CreateAdapterResponse$ = [3, n0, _CARr,
    0,
    [_AI],
    [0]
];
var CreateAdapterVersionRequest$ = [3, n0, _CAVR,
    0,
    [_AI, _DC, _OC, _CRT, _KMSKI, _Ta],
    [0, () => AdapterVersionDatasetConfig$, () => OutputConfig$, [0, 4], 0, 128 | 0], 3
];
var CreateAdapterVersionResponse$ = [3, n0, _CAVRr,
    0,
    [_AI, _AV],
    [0, 0]
];
var DeleteAdapterRequest$ = [3, n0, _DAR,
    0,
    [_AI],
    [0], 1
];
var DeleteAdapterResponse$ = [3, n0, _DARe,
    0,
    [],
    []
];
var DeleteAdapterVersionRequest$ = [3, n0, _DAVR,
    0,
    [_AI, _AV],
    [0, 0], 2
];
var DeleteAdapterVersionResponse$ = [3, n0, _DAVRe,
    0,
    [],
    []
];
var DetectDocumentTextRequest$ = [3, n0, _DDTR,
    0,
    [_D],
    [() => Document$], 1
];
var DetectDocumentTextResponse$ = [3, n0, _DDTRe,
    0,
    [_DM, _Bl, _DDTMV],
    [() => DocumentMetadata$, () => BlockList, 0]
];
var DetectedSignature$ = [3, n0, _DS,
    0,
    [_Pa],
    [1]
];
var Document$ = [3, n0, _D,
    0,
    [_By, _SO],
    [21, () => S3Object$]
];
var DocumentGroup$ = [3, n0, _DG,
    0,
    [_Ty, _SD, _DSe, _US],
    [0, () => SplitDocumentList, () => DetectedSignatureList, () => UndetectedSignatureList]
];
var DocumentLocation$ = [3, n0, _DL,
    0,
    [_SO],
    [() => S3Object$]
];
var DocumentMetadata$ = [3, n0, _DM,
    0,
    [_P],
    [1]
];
var EvaluationMetric$ = [3, n0, _EM,
    0,
    [_FS, _Pr, _Re],
    [1, 1, 1]
];
var ExpenseCurrency$ = [3, n0, _EC,
    0,
    [_C, _Co],
    [0, 1]
];
var ExpenseDetection$ = [3, n0, _EDx,
    0,
    [_T, _G, _Co],
    [0, () => Geometry$, 1]
];
var ExpenseDocument$ = [3, n0, _EDxp,
    0,
    [_EI, _SF, _LIG, _Bl],
    [1, () => ExpenseFieldList, () => LineItemGroupList, () => BlockList]
];
var ExpenseField$ = [3, n0, _EF,
    0,
    [_Ty, _LD, _VD, _PN, _Cu, _GP],
    [() => ExpenseType$, () => ExpenseDetection$, () => ExpenseDetection$, 1, () => ExpenseCurrency$, () => ExpenseGroupPropertyList]
];
var ExpenseGroupProperty$ = [3, n0, _EGP,
    0,
    [_Typ, _I],
    [64 | 0, 0]
];
var ExpenseType$ = [3, n0, _ETx,
    0,
    [_T, _Co],
    [0, 1]
];
var Extraction$ = [3, n0, _E,
    0,
    [_LDe, _EDxp, _IDd],
    [() => LendingDocument$, () => ExpenseDocument$, () => IdentityDocument$]
];
var Geometry$ = [3, n0, _G,
    0,
    [_BB, _Po, _RA],
    [() => BoundingBox$, () => Polygon, 1]
];
var GetAdapterRequest$ = [3, n0, _GAR,
    0,
    [_AI],
    [0], 1
];
var GetAdapterResponse$ = [3, n0, _GARe,
    0,
    [_AI, _AN, _CT, _De, _FT, _AU, _Ta],
    [0, 0, 4, 0, 64 | 0, 0, 128 | 0]
];
var GetAdapterVersionRequest$ = [3, n0, _GAVR,
    0,
    [_AI, _AV],
    [0, 0], 2
];
var GetAdapterVersionResponse$ = [3, n0, _GAVRe,
    0,
    [_AI, _AV, _CT, _FT, _S, _SM, _DC, _KMSKI, _OC, _EMv, _Ta],
    [0, 0, 4, 64 | 0, 0, 0, () => AdapterVersionDatasetConfig$, 0, () => OutputConfig$, () => AdapterVersionEvaluationMetrics, 128 | 0]
];
var GetDocumentAnalysisRequest$ = [3, n0, _GDAR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
var GetDocumentAnalysisResponse$ = [3, n0, _GDARe,
    0,
    [_DM, _JS, _NT, _Bl, _Wa, _SM, _ADMV],
    [() => DocumentMetadata$, 0, 0, () => BlockList, () => Warnings, 0, 0]
];
var GetDocumentTextDetectionRequest$ = [3, n0, _GDTDR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
var GetDocumentTextDetectionResponse$ = [3, n0, _GDTDRe,
    0,
    [_DM, _JS, _NT, _Bl, _Wa, _SM, _DDTMV],
    [() => DocumentMetadata$, 0, 0, () => BlockList, () => Warnings, 0, 0]
];
var GetExpenseAnalysisRequest$ = [3, n0, _GEAR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
var GetExpenseAnalysisResponse$ = [3, n0, _GEARe,
    0,
    [_DM, _JS, _NT, _ED, _Wa, _SM, _AEMV],
    [() => DocumentMetadata$, 0, 0, () => ExpenseDocumentList, () => Warnings, 0, 0]
];
var GetLendingAnalysisRequest$ = [3, n0, _GLAR,
    0,
    [_JI, _MR, _NT],
    [0, 1, 0], 1
];
var GetLendingAnalysisResponse$ = [3, n0, _GLARe,
    0,
    [_DM, _JS, _NT, _Res, _Wa, _SM, _ALMV],
    [() => DocumentMetadata$, 0, 0, () => LendingResultList, () => Warnings, 0, 0]
];
var GetLendingAnalysisSummaryRequest$ = [3, n0, _GLASR,
    0,
    [_JI],
    [0], 1
];
var GetLendingAnalysisSummaryResponse$ = [3, n0, _GLASRe,
    0,
    [_DM, _JS, _Su, _Wa, _SM, _ALMV],
    [() => DocumentMetadata$, 0, () => LendingSummary$, () => Warnings, 0, 0]
];
var HumanLoopActivationOutput$ = [3, n0, _HLAO,
    0,
    [_HLA, _HLAR, _HLACER],
    [0, 64 | 0, [() => SynthesizedJsonHumanLoopActivationConditionsEvaluationResults, 0]]
];
var HumanLoopConfig$ = [3, n0, _HLC,
    0,
    [_HLN, _FDA, _DA],
    [0, 0, () => HumanLoopDataAttributes$], 2
];
var HumanLoopDataAttributes$ = [3, n0, _HLDA,
    0,
    [_CC],
    [64 | 0]
];
var IdentityDocument$ = [3, n0, _IDd,
    0,
    [_DI, _IDF, _Bl],
    [1, () => IdentityDocumentFieldList, () => BlockList]
];
var IdentityDocumentField$ = [3, n0, _IDFd,
    0,
    [_Ty, _VD],
    [() => AnalyzeIDDetections$, () => AnalyzeIDDetections$]
];
var LendingDetection$ = [3, n0, _LDen,
    0,
    [_T, _SS, _G, _Co],
    [0, 0, () => Geometry$, 1]
];
var LendingDocument$ = [3, n0, _LDe,
    0,
    [_LF, _SDi],
    [() => LendingFieldList, () => SignatureDetectionList]
];
var LendingField$ = [3, n0, _LFe,
    0,
    [_Ty, _KD, _VDa],
    [0, () => LendingDetection$, () => LendingDetectionList]
];
var LendingResult$ = [3, n0, _LR,
    0,
    [_Pa, _PC, _Ex],
    [1, () => PageClassification$, () => ExtractionList]
];
var LendingSummary$ = [3, n0, _LS,
    0,
    [_DGo, _UDT],
    [() => DocumentGroupList, 64 | 0]
];
var LineItemFields$ = [3, n0, _LIF,
    0,
    [_LIEF],
    [() => ExpenseFieldList]
];
var LineItemGroup$ = [3, n0, _LIGi,
    0,
    [_LIGI, _LI],
    [1, () => LineItemList]
];
var ListAdaptersRequest$ = [3, n0, _LAR,
    0,
    [_ACT, _BCT, _MR, _NT],
    [4, 4, 1, 0]
];
var ListAdaptersResponse$ = [3, n0, _LARi,
    0,
    [_Ad, _NT],
    [() => AdapterList, 0]
];
var ListAdapterVersionsRequest$ = [3, n0, _LAVR,
    0,
    [_AI, _ACT, _BCT, _MR, _NT],
    [0, 4, 4, 1, 0]
];
var ListAdapterVersionsResponse$ = [3, n0, _LAVRi,
    0,
    [_AVd, _NT],
    [() => AdapterVersionList, 0]
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [128 | 0]
];
var NormalizedValue$ = [3, n0, _NV,
    0,
    [_Va, _VT],
    [0, 0]
];
var NotificationChannel$ = [3, n0, _NC,
    0,
    [_SNSTA, _RAo],
    [0, 0], 2
];
var OutputConfig$ = [3, n0, _OC,
    0,
    [_SB, _SP],
    [0, 0], 1
];
var PageClassification$ = [3, n0, _PC,
    0,
    [_PT, _PN],
    [() => PredictionList, () => PredictionList], 2
];
var Point$ = [3, n0, _Poi,
    0,
    [_X, _Y],
    [1, 1]
];
var Prediction$ = [3, n0, _Pre,
    0,
    [_Va, _Co],
    [0, 1]
];
var QueriesConfig$ = [3, n0, _QCu,
    0,
    [_Qu],
    [() => Queries], 1
];
var Query$ = [3, n0, _Q,
    0,
    [_T, _Al, _P],
    [0, 0, 64 | 0], 1
];
var Relationship$ = [3, n0, _Rel,
    0,
    [_Ty, _Id],
    [0, 64 | 0]
];
var S3Object$ = [3, n0, _SO,
    0,
    [_Bu, _N, _V],
    [0, 0, 0]
];
var SignatureDetection$ = [3, n0, _SDig,
    0,
    [_Co, _G],
    [1, () => Geometry$]
];
var SplitDocument$ = [3, n0, _SDp,
    0,
    [_In, _P],
    [1, 64 | 1]
];
var StartDocumentAnalysisRequest$ = [3, n0, _SDAR,
    0,
    [_DL, _FT, _CRT, _JT, _NC, _OC, _KMSKI, _QCu, _AC],
    [() => DocumentLocation$, 64 | 0, 0, 0, () => NotificationChannel$, () => OutputConfig$, 0, () => QueriesConfig$, () => AdaptersConfig$], 2
];
var StartDocumentAnalysisResponse$ = [3, n0, _SDARt,
    0,
    [_JI],
    [0]
];
var StartDocumentTextDetectionRequest$ = [3, n0, _SDTDR,
    0,
    [_DL, _CRT, _JT, _NC, _OC, _KMSKI],
    [() => DocumentLocation$, 0, 0, () => NotificationChannel$, () => OutputConfig$, 0], 1
];
var StartDocumentTextDetectionResponse$ = [3, n0, _SDTDRt,
    0,
    [_JI],
    [0]
];
var StartExpenseAnalysisRequest$ = [3, n0, _SEAR,
    0,
    [_DL, _CRT, _JT, _NC, _OC, _KMSKI],
    [() => DocumentLocation$, 0, 0, () => NotificationChannel$, () => OutputConfig$, 0], 1
];
var StartExpenseAnalysisResponse$ = [3, n0, _SEARt,
    0,
    [_JI],
    [0]
];
var StartLendingAnalysisRequest$ = [3, n0, _SLAR,
    0,
    [_DL, _CRT, _JT, _NC, _OC, _KMSKI],
    [() => DocumentLocation$, 0, 0, () => NotificationChannel$, () => OutputConfig$, 0], 1
];
var StartLendingAnalysisResponse$ = [3, n0, _SLARt,
    0,
    [_JI],
    [0]
];
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _Ta],
    [0, 128 | 0], 2
];
var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
var UndetectedSignature$ = [3, n0, _USn,
    0,
    [_Pa],
    [1]
];
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
var UpdateAdapterRequest$ = [3, n0, _UAR,
    0,
    [_AI, _De, _AN, _AU],
    [0, 0, 0, 0], 1
];
var UpdateAdapterResponse$ = [3, n0, _UARp,
    0,
    [_AI, _AN, _CT, _De, _FT, _AU],
    [0, 0, 4, 0, 64 | 0, 0]
];
var Warning$ = [3, n0, _War,
    0,
    [_ECr, _P],
    [0, 64 | 1]
];
var AdapterList = [1, n0, _AL,
    0, () => AdapterOverview$
];
var Adapters = [1, n0, _Ad,
    0, () => Adapter$
];
var AdapterVersionEvaluationMetrics = [1, n0, _AVEMd,
    0, () => AdapterVersionEvaluationMetric$
];
var AdapterVersionList = [1, n0, _AVL,
    0, () => AdapterVersionOverview$
];
var BlockList = [1, n0, _BL,
    0, () => Block$
];
var DetectedSignatureList = [1, n0, _DSL,
    0, () => DetectedSignature$
];
var DocumentGroupList = [1, n0, _DGL,
    0, () => DocumentGroup$
];
var DocumentPages = [1, n0, _DP,
    0, () => Document$
];
var ExpenseDocumentList = [1, n0, _EDL,
    0, () => ExpenseDocument$
];
var ExpenseFieldList = [1, n0, _EFL,
    0, () => ExpenseField$
];
var ExpenseGroupPropertyList = [1, n0, _EGPL,
    0, () => ExpenseGroupProperty$
];
var ExtractionList = [1, n0, _EL,
    0, () => Extraction$
];
var IdentityDocumentFieldList = [1, n0, _IDFL,
    0, () => IdentityDocumentField$
];
var IdentityDocumentList = [1, n0, _IDL,
    0, () => IdentityDocument$
];
var LendingDetectionList = [1, n0, _LDL,
    0, () => LendingDetection$
];
var LendingFieldList = [1, n0, _LFL,
    0, () => LendingField$
];
var LendingResultList = [1, n0, _LRL,
    0, () => LendingResult$
];
var LineItemGroupList = [1, n0, _LIGL,
    0, () => LineItemGroup$
];
var LineItemList = [1, n0, _LIL,
    0, () => LineItemFields$
];
var Polygon = [1, n0, _Po,
    0, () => Point$
];
var PredictionList = [1, n0, _PL,
    0, () => Prediction$
];
var Queries = [1, n0, _Qu,
    0, () => Query$
];
var RelationshipList = [1, n0, _RL,
    0, () => Relationship$
];
var SignatureDetectionList = [1, n0, _SDL,
    0, () => SignatureDetection$
];
var SplitDocumentList = [1, n0, _SDLp,
    0, () => SplitDocument$
];
var UndetectedSignatureList = [1, n0, _USL,
    0, () => UndetectedSignature$
];
var Warnings = [1, n0, _Wa,
    0, () => Warning$
];
var AnalyzeDocument$ = [9, n0, _AD,
    0, () => AnalyzeDocumentRequest$, () => AnalyzeDocumentResponse$
];
var AnalyzeExpense$ = [9, n0, _AE,
    0, () => AnalyzeExpenseRequest$, () => AnalyzeExpenseResponse$
];
var AnalyzeID$ = [9, n0, _AID,
    0, () => AnalyzeIDRequest$, () => AnalyzeIDResponse$
];
var CreateAdapter$ = [9, n0, _CA,
    2, () => CreateAdapterRequest$, () => CreateAdapterResponse$
];
var CreateAdapterVersion$ = [9, n0, _CAV,
    2, () => CreateAdapterVersionRequest$, () => CreateAdapterVersionResponse$
];
var DeleteAdapter$ = [9, n0, _DAe,
    2, () => DeleteAdapterRequest$, () => DeleteAdapterResponse$
];
var DeleteAdapterVersion$ = [9, n0, _DAV,
    2, () => DeleteAdapterVersionRequest$, () => DeleteAdapterVersionResponse$
];
var DetectDocumentText$ = [9, n0, _DDT,
    0, () => DetectDocumentTextRequest$, () => DetectDocumentTextResponse$
];
var GetAdapter$ = [9, n0, _GA,
    0, () => GetAdapterRequest$, () => GetAdapterResponse$
];
var GetAdapterVersion$ = [9, n0, _GAV,
    0, () => GetAdapterVersionRequest$, () => GetAdapterVersionResponse$
];
var GetDocumentAnalysis$ = [9, n0, _GDA,
    0, () => GetDocumentAnalysisRequest$, () => GetDocumentAnalysisResponse$
];
var GetDocumentTextDetection$ = [9, n0, _GDTD,
    0, () => GetDocumentTextDetectionRequest$, () => GetDocumentTextDetectionResponse$
];
var GetExpenseAnalysis$ = [9, n0, _GEA,
    0, () => GetExpenseAnalysisRequest$, () => GetExpenseAnalysisResponse$
];
var GetLendingAnalysis$ = [9, n0, _GLA,
    0, () => GetLendingAnalysisRequest$, () => GetLendingAnalysisResponse$
];
var GetLendingAnalysisSummary$ = [9, n0, _GLAS,
    0, () => GetLendingAnalysisSummaryRequest$, () => GetLendingAnalysisSummaryResponse$
];
var ListAdapters$ = [9, n0, _LA,
    0, () => ListAdaptersRequest$, () => ListAdaptersResponse$
];
var ListAdapterVersions$ = [9, n0, _LAV,
    0, () => ListAdapterVersionsRequest$, () => ListAdapterVersionsResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var StartDocumentAnalysis$ = [9, n0, _SDA,
    0, () => StartDocumentAnalysisRequest$, () => StartDocumentAnalysisResponse$
];
var StartDocumentTextDetection$ = [9, n0, _SDTD,
    0, () => StartDocumentTextDetectionRequest$, () => StartDocumentTextDetectionResponse$
];
var StartExpenseAnalysis$ = [9, n0, _SEA,
    0, () => StartExpenseAnalysisRequest$, () => StartExpenseAnalysisResponse$
];
var StartLendingAnalysis$ = [9, n0, _SLA,
    0, () => StartLendingAnalysisRequest$, () => StartLendingAnalysisResponse$
];
var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
var UpdateAdapter$ = [9, n0, _UA,
    0, () => UpdateAdapterRequest$, () => UpdateAdapterResponse$
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2018-06-27",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultTextractHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsJson1_1Protocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.textract",
            errorTypeRegistries,
            version: "2018-06-27",
            serviceTarget: "Textract",
        },
        serviceId: config?.serviceId ?? "Textract",
        sha256: config?.sha256 ?? Sha256,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class TextractClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultTextractHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "Textract", "TextractClient", getEndpointPlugin);
const _ep0 = {};
const _mw0 = (Command, cs, config, o) => [];

class AnalyzeDocumentCommand extends command(_ep0, _mw0, "AnalyzeDocument", AnalyzeDocument$) {
}

class AnalyzeExpenseCommand extends command(_ep0, _mw0, "AnalyzeExpense", AnalyzeExpense$) {
}

class AnalyzeIDCommand extends command(_ep0, _mw0, "AnalyzeID", AnalyzeID$) {
}

class CreateAdapterCommand extends command(_ep0, _mw0, "CreateAdapter", CreateAdapter$) {
}

class CreateAdapterVersionCommand extends command(_ep0, _mw0, "CreateAdapterVersion", CreateAdapterVersion$) {
}

class DeleteAdapterCommand extends command(_ep0, _mw0, "DeleteAdapter", DeleteAdapter$) {
}

class DeleteAdapterVersionCommand extends command(_ep0, _mw0, "DeleteAdapterVersion", DeleteAdapterVersion$) {
}

class DetectDocumentTextCommand extends command(_ep0, _mw0, "DetectDocumentText", DetectDocumentText$) {
}

class GetAdapterCommand extends command(_ep0, _mw0, "GetAdapter", GetAdapter$) {
}

class GetAdapterVersionCommand extends command(_ep0, _mw0, "GetAdapterVersion", GetAdapterVersion$) {
}

class GetDocumentAnalysisCommand extends command(_ep0, _mw0, "GetDocumentAnalysis", GetDocumentAnalysis$) {
}

class GetDocumentTextDetectionCommand extends command(_ep0, _mw0, "GetDocumentTextDetection", GetDocumentTextDetection$) {
}

class GetExpenseAnalysisCommand extends command(_ep0, _mw0, "GetExpenseAnalysis", GetExpenseAnalysis$) {
}

class GetLendingAnalysisCommand extends command(_ep0, _mw0, "GetLendingAnalysis", GetLendingAnalysis$) {
}

class GetLendingAnalysisSummaryCommand extends command(_ep0, _mw0, "GetLendingAnalysisSummary", GetLendingAnalysisSummary$) {
}

class ListAdaptersCommand extends command(_ep0, _mw0, "ListAdapters", ListAdapters$) {
}

class ListAdapterVersionsCommand extends command(_ep0, _mw0, "ListAdapterVersions", ListAdapterVersions$) {
}

class ListTagsForResourceCommand extends command(_ep0, _mw0, "ListTagsForResource", ListTagsForResource$) {
}

class StartDocumentAnalysisCommand extends command(_ep0, _mw0, "StartDocumentAnalysis", StartDocumentAnalysis$) {
}

class StartDocumentTextDetectionCommand extends command(_ep0, _mw0, "StartDocumentTextDetection", StartDocumentTextDetection$) {
}

class StartExpenseAnalysisCommand extends command(_ep0, _mw0, "StartExpenseAnalysis", StartExpenseAnalysis$) {
}

class StartLendingAnalysisCommand extends command(_ep0, _mw0, "StartLendingAnalysis", StartLendingAnalysis$) {
}

class TagResourceCommand extends command(_ep0, _mw0, "TagResource", TagResource$) {
}

class UntagResourceCommand extends command(_ep0, _mw0, "UntagResource", UntagResource$) {
}

class UpdateAdapterCommand extends command(_ep0, _mw0, "UpdateAdapter", UpdateAdapter$) {
}

const paginateListAdapters = createPaginator(TextractClient, ListAdaptersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListAdapterVersions = createPaginator(TextractClient, ListAdapterVersionsCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AnalyzeDocumentCommand,
    AnalyzeExpenseCommand,
    AnalyzeIDCommand,
    CreateAdapterCommand,
    CreateAdapterVersionCommand,
    DeleteAdapterCommand,
    DeleteAdapterVersionCommand,
    DetectDocumentTextCommand,
    GetAdapterCommand,
    GetAdapterVersionCommand,
    GetDocumentAnalysisCommand,
    GetDocumentTextDetectionCommand,
    GetExpenseAnalysisCommand,
    GetLendingAnalysisCommand,
    GetLendingAnalysisSummaryCommand,
    ListAdaptersCommand,
    ListAdapterVersionsCommand,
    ListTagsForResourceCommand,
    StartDocumentAnalysisCommand,
    StartDocumentTextDetectionCommand,
    StartExpenseAnalysisCommand,
    StartLendingAnalysisCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAdapterCommand,
};
const paginators = {
    paginateListAdapters,
    paginateListAdapterVersions,
};
class Textract extends TextractClient {
}
createAggregatedClient(commands, Textract, { paginators });

const FeatureType = {
    FORMS: "FORMS",
    LAYOUT: "LAYOUT",
    QUERIES: "QUERIES",
    SIGNATURES: "SIGNATURES",
    TABLES: "TABLES",
};
const AdapterVersionStatus = {
    ACTIVE: "ACTIVE",
    AT_RISK: "AT_RISK",
    CREATION_ERROR: "CREATION_ERROR",
    CREATION_IN_PROGRESS: "CREATION_IN_PROGRESS",
    DEPRECATED: "DEPRECATED",
};
const ContentClassifier = {
    FREE_OF_ADULT_CONTENT: "FreeOfAdultContent",
    FREE_OF_PERSONALLY_IDENTIFIABLE_INFORMATION: "FreeOfPersonallyIdentifiableInformation",
};
const BlockType = {
    CELL: "CELL",
    KEY_VALUE_SET: "KEY_VALUE_SET",
    LAYOUT_FIGURE: "LAYOUT_FIGURE",
    LAYOUT_FOOTER: "LAYOUT_FOOTER",
    LAYOUT_HEADER: "LAYOUT_HEADER",
    LAYOUT_KEY_VALUE: "LAYOUT_KEY_VALUE",
    LAYOUT_LIST: "LAYOUT_LIST",
    LAYOUT_PAGE_NUMBER: "LAYOUT_PAGE_NUMBER",
    LAYOUT_SECTION_HEADER: "LAYOUT_SECTION_HEADER",
    LAYOUT_TABLE: "LAYOUT_TABLE",
    LAYOUT_TEXT: "LAYOUT_TEXT",
    LAYOUT_TITLE: "LAYOUT_TITLE",
    LINE: "LINE",
    MERGED_CELL: "MERGED_CELL",
    PAGE: "PAGE",
    QUERY: "QUERY",
    QUERY_RESULT: "QUERY_RESULT",
    SELECTION_ELEMENT: "SELECTION_ELEMENT",
    SIGNATURE: "SIGNATURE",
    TABLE: "TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_TITLE: "TABLE_TITLE",
    TITLE: "TITLE",
    WORD: "WORD",
};
const EntityType = {
    COLUMN_HEADER: "COLUMN_HEADER",
    KEY: "KEY",
    SEMI_STRUCTURED_TABLE: "SEMI_STRUCTURED_TABLE",
    STRUCTURED_TABLE: "STRUCTURED_TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_SECTION_TITLE: "TABLE_SECTION_TITLE",
    TABLE_SUMMARY: "TABLE_SUMMARY",
    TABLE_TITLE: "TABLE_TITLE",
    VALUE: "VALUE",
};
const RelationshipType = {
    ANSWER: "ANSWER",
    CHILD: "CHILD",
    COMPLEX_FEATURES: "COMPLEX_FEATURES",
    MERGED_CELL: "MERGED_CELL",
    TABLE: "TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_TITLE: "TABLE_TITLE",
    TITLE: "TITLE",
    VALUE: "VALUE",
};
const SelectionStatus = {
    NOT_SELECTED: "NOT_SELECTED",
    SELECTED: "SELECTED",
};
const TextType = {
    HANDWRITING: "HANDWRITING",
    PRINTED: "PRINTED",
};
const ValueType = {
    DATE: "DATE",
};
const AutoUpdate = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const JobStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PARTIAL_SUCCESS: "PARTIAL_SUCCESS",
    SUCCEEDED: "SUCCEEDED",
};

exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.Adapter$ = Adapter$;
exports.AdapterOverview$ = AdapterOverview$;
exports.AdapterVersionDatasetConfig$ = AdapterVersionDatasetConfig$;
exports.AdapterVersionEvaluationMetric$ = AdapterVersionEvaluationMetric$;
exports.AdapterVersionOverview$ = AdapterVersionOverview$;
exports.AdapterVersionStatus = AdapterVersionStatus;
exports.AdaptersConfig$ = AdaptersConfig$;
exports.AnalyzeDocument$ = AnalyzeDocument$;
exports.AnalyzeDocumentCommand = AnalyzeDocumentCommand;
exports.AnalyzeDocumentRequest$ = AnalyzeDocumentRequest$;
exports.AnalyzeDocumentResponse$ = AnalyzeDocumentResponse$;
exports.AnalyzeExpense$ = AnalyzeExpense$;
exports.AnalyzeExpenseCommand = AnalyzeExpenseCommand;
exports.AnalyzeExpenseRequest$ = AnalyzeExpenseRequest$;
exports.AnalyzeExpenseResponse$ = AnalyzeExpenseResponse$;
exports.AnalyzeID$ = AnalyzeID$;
exports.AnalyzeIDCommand = AnalyzeIDCommand;
exports.AnalyzeIDDetections$ = AnalyzeIDDetections$;
exports.AnalyzeIDRequest$ = AnalyzeIDRequest$;
exports.AnalyzeIDResponse$ = AnalyzeIDResponse$;
exports.AutoUpdate = AutoUpdate;
exports.BadDocumentException = BadDocumentException;
exports.BadDocumentException$ = BadDocumentException$;
exports.Block$ = Block$;
exports.BlockType = BlockType;
exports.BoundingBox$ = BoundingBox$;
exports.ConflictException = ConflictException;
exports.ConflictException$ = ConflictException$;
exports.ContentClassifier = ContentClassifier;
exports.CreateAdapter$ = CreateAdapter$;
exports.CreateAdapterCommand = CreateAdapterCommand;
exports.CreateAdapterRequest$ = CreateAdapterRequest$;
exports.CreateAdapterResponse$ = CreateAdapterResponse$;
exports.CreateAdapterVersion$ = CreateAdapterVersion$;
exports.CreateAdapterVersionCommand = CreateAdapterVersionCommand;
exports.CreateAdapterVersionRequest$ = CreateAdapterVersionRequest$;
exports.CreateAdapterVersionResponse$ = CreateAdapterVersionResponse$;
exports.DeleteAdapter$ = DeleteAdapter$;
exports.DeleteAdapterCommand = DeleteAdapterCommand;
exports.DeleteAdapterRequest$ = DeleteAdapterRequest$;
exports.DeleteAdapterResponse$ = DeleteAdapterResponse$;
exports.DeleteAdapterVersion$ = DeleteAdapterVersion$;
exports.DeleteAdapterVersionCommand = DeleteAdapterVersionCommand;
exports.DeleteAdapterVersionRequest$ = DeleteAdapterVersionRequest$;
exports.DeleteAdapterVersionResponse$ = DeleteAdapterVersionResponse$;
exports.DetectDocumentText$ = DetectDocumentText$;
exports.DetectDocumentTextCommand = DetectDocumentTextCommand;
exports.DetectDocumentTextRequest$ = DetectDocumentTextRequest$;
exports.DetectDocumentTextResponse$ = DetectDocumentTextResponse$;
exports.DetectedSignature$ = DetectedSignature$;
exports.Document$ = Document$;
exports.DocumentGroup$ = DocumentGroup$;
exports.DocumentLocation$ = DocumentLocation$;
exports.DocumentMetadata$ = DocumentMetadata$;
exports.DocumentTooLargeException = DocumentTooLargeException;
exports.DocumentTooLargeException$ = DocumentTooLargeException$;
exports.EntityType = EntityType;
exports.EvaluationMetric$ = EvaluationMetric$;
exports.ExpenseCurrency$ = ExpenseCurrency$;
exports.ExpenseDetection$ = ExpenseDetection$;
exports.ExpenseDocument$ = ExpenseDocument$;
exports.ExpenseField$ = ExpenseField$;
exports.ExpenseGroupProperty$ = ExpenseGroupProperty$;
exports.ExpenseType$ = ExpenseType$;
exports.Extraction$ = Extraction$;
exports.FeatureType = FeatureType;
exports.Geometry$ = Geometry$;
exports.GetAdapter$ = GetAdapter$;
exports.GetAdapterCommand = GetAdapterCommand;
exports.GetAdapterRequest$ = GetAdapterRequest$;
exports.GetAdapterResponse$ = GetAdapterResponse$;
exports.GetAdapterVersion$ = GetAdapterVersion$;
exports.GetAdapterVersionCommand = GetAdapterVersionCommand;
exports.GetAdapterVersionRequest$ = GetAdapterVersionRequest$;
exports.GetAdapterVersionResponse$ = GetAdapterVersionResponse$;
exports.GetDocumentAnalysis$ = GetDocumentAnalysis$;
exports.GetDocumentAnalysisCommand = GetDocumentAnalysisCommand;
exports.GetDocumentAnalysisRequest$ = GetDocumentAnalysisRequest$;
exports.GetDocumentAnalysisResponse$ = GetDocumentAnalysisResponse$;
exports.GetDocumentTextDetection$ = GetDocumentTextDetection$;
exports.GetDocumentTextDetectionCommand = GetDocumentTextDetectionCommand;
exports.GetDocumentTextDetectionRequest$ = GetDocumentTextDetectionRequest$;
exports.GetDocumentTextDetectionResponse$ = GetDocumentTextDetectionResponse$;
exports.GetExpenseAnalysis$ = GetExpenseAnalysis$;
exports.GetExpenseAnalysisCommand = GetExpenseAnalysisCommand;
exports.GetExpenseAnalysisRequest$ = GetExpenseAnalysisRequest$;
exports.GetExpenseAnalysisResponse$ = GetExpenseAnalysisResponse$;
exports.GetLendingAnalysis$ = GetLendingAnalysis$;
exports.GetLendingAnalysisCommand = GetLendingAnalysisCommand;
exports.GetLendingAnalysisRequest$ = GetLendingAnalysisRequest$;
exports.GetLendingAnalysisResponse$ = GetLendingAnalysisResponse$;
exports.GetLendingAnalysisSummary$ = GetLendingAnalysisSummary$;
exports.GetLendingAnalysisSummaryCommand = GetLendingAnalysisSummaryCommand;
exports.GetLendingAnalysisSummaryRequest$ = GetLendingAnalysisSummaryRequest$;
exports.GetLendingAnalysisSummaryResponse$ = GetLendingAnalysisSummaryResponse$;
exports.HumanLoopActivationOutput$ = HumanLoopActivationOutput$;
exports.HumanLoopConfig$ = HumanLoopConfig$;
exports.HumanLoopDataAttributes$ = HumanLoopDataAttributes$;
exports.HumanLoopQuotaExceededException = HumanLoopQuotaExceededException;
exports.HumanLoopQuotaExceededException$ = HumanLoopQuotaExceededException$;
exports.IdempotentParameterMismatchException = IdempotentParameterMismatchException;
exports.IdempotentParameterMismatchException$ = IdempotentParameterMismatchException$;
exports.IdentityDocument$ = IdentityDocument$;
exports.IdentityDocumentField$ = IdentityDocumentField$;
exports.InternalServerError = InternalServerError;
exports.InternalServerError$ = InternalServerError$;
exports.InvalidJobIdException = InvalidJobIdException;
exports.InvalidJobIdException$ = InvalidJobIdException$;
exports.InvalidKMSKeyException = InvalidKMSKeyException;
exports.InvalidKMSKeyException$ = InvalidKMSKeyException$;
exports.InvalidParameterException = InvalidParameterException;
exports.InvalidParameterException$ = InvalidParameterException$;
exports.InvalidS3ObjectException = InvalidS3ObjectException;
exports.InvalidS3ObjectException$ = InvalidS3ObjectException$;
exports.JobStatus = JobStatus;
exports.LendingDetection$ = LendingDetection$;
exports.LendingDocument$ = LendingDocument$;
exports.LendingField$ = LendingField$;
exports.LendingResult$ = LendingResult$;
exports.LendingSummary$ = LendingSummary$;
exports.LimitExceededException = LimitExceededException;
exports.LimitExceededException$ = LimitExceededException$;
exports.LineItemFields$ = LineItemFields$;
exports.LineItemGroup$ = LineItemGroup$;
exports.ListAdapterVersions$ = ListAdapterVersions$;
exports.ListAdapterVersionsCommand = ListAdapterVersionsCommand;
exports.ListAdapterVersionsRequest$ = ListAdapterVersionsRequest$;
exports.ListAdapterVersionsResponse$ = ListAdapterVersionsResponse$;
exports.ListAdapters$ = ListAdapters$;
exports.ListAdaptersCommand = ListAdaptersCommand;
exports.ListAdaptersRequest$ = ListAdaptersRequest$;
exports.ListAdaptersResponse$ = ListAdaptersResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.NormalizedValue$ = NormalizedValue$;
exports.NotificationChannel$ = NotificationChannel$;
exports.OutputConfig$ = OutputConfig$;
exports.PageClassification$ = PageClassification$;
exports.Point$ = Point$;
exports.Prediction$ = Prediction$;
exports.ProvisionedThroughputExceededException = ProvisionedThroughputExceededException;
exports.ProvisionedThroughputExceededException$ = ProvisionedThroughputExceededException$;
exports.QueriesConfig$ = QueriesConfig$;
exports.Query$ = Query$;
exports.Relationship$ = Relationship$;
exports.RelationshipType = RelationshipType;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.S3Object$ = S3Object$;
exports.SelectionStatus = SelectionStatus;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
exports.ServiceQuotaExceededException$ = ServiceQuotaExceededException$;
exports.SignatureDetection$ = SignatureDetection$;
exports.SplitDocument$ = SplitDocument$;
exports.StartDocumentAnalysis$ = StartDocumentAnalysis$;
exports.StartDocumentAnalysisCommand = StartDocumentAnalysisCommand;
exports.StartDocumentAnalysisRequest$ = StartDocumentAnalysisRequest$;
exports.StartDocumentAnalysisResponse$ = StartDocumentAnalysisResponse$;
exports.StartDocumentTextDetection$ = StartDocumentTextDetection$;
exports.StartDocumentTextDetectionCommand = StartDocumentTextDetectionCommand;
exports.StartDocumentTextDetectionRequest$ = StartDocumentTextDetectionRequest$;
exports.StartDocumentTextDetectionResponse$ = StartDocumentTextDetectionResponse$;
exports.StartExpenseAnalysis$ = StartExpenseAnalysis$;
exports.StartExpenseAnalysisCommand = StartExpenseAnalysisCommand;
exports.StartExpenseAnalysisRequest$ = StartExpenseAnalysisRequest$;
exports.StartExpenseAnalysisResponse$ = StartExpenseAnalysisResponse$;
exports.StartLendingAnalysis$ = StartLendingAnalysis$;
exports.StartLendingAnalysisCommand = StartLendingAnalysisCommand;
exports.StartLendingAnalysisRequest$ = StartLendingAnalysisRequest$;
exports.StartLendingAnalysisResponse$ = StartLendingAnalysisResponse$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.TagResourceResponse$ = TagResourceResponse$;
exports.TextType = TextType;
exports.Textract = Textract;
exports.TextractClient = TextractClient;
exports.TextractServiceException = TextractServiceException;
exports.TextractServiceException$ = TextractServiceException$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.UndetectedSignature$ = UndetectedSignature$;
exports.UnsupportedDocumentException = UnsupportedDocumentException;
exports.UnsupportedDocumentException$ = UnsupportedDocumentException$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UntagResourceResponse$ = UntagResourceResponse$;
exports.UpdateAdapter$ = UpdateAdapter$;
exports.UpdateAdapterCommand = UpdateAdapterCommand;
exports.UpdateAdapterRequest$ = UpdateAdapterRequest$;
exports.UpdateAdapterResponse$ = UpdateAdapterResponse$;
exports.ValidationException = ValidationException;
exports.ValidationException$ = ValidationException$;
exports.ValueType = ValueType;
exports.Warning$ = Warning$;
exports.errorTypeRegistries = errorTypeRegistries;
exports.paginateListAdapterVersions = paginateListAdapterVersions;
exports.paginateListAdapters = paginateListAdapters;
